File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int n = prices.length;
        int indexBuy = 0;
        for ( int i = 0; i < n; i++){
            if(prices[i] < prices[indexBuy]){
                indexBuy = i;
            }
            maxProfit = Math.max(maxProfit, prices[i] - prices[indexBuy]);
        }
        
        return maxProfit;
        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        l, r = 0, 1
        while l < r and r <= len(prices) - 1:
            if prices[l] >= prices[r]:
                l = r
                r = r + 1
            else:
                profit = prices[r] - prices[l]
                res = max(res, profit)
                r = r + 1
        return res
        